library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -39,15 +41,16 @@ data class ClientOptions(
     )
 }
 
+@JvmInline
+value class InboxId(val value: String)
```

Comment:
I made this a value class so that it would error if people pass an address instead of an inboxId into certain methods. If you pass an address instead of a inboxId you get this error: https://github.com/xmtp/libxmtp/issues/1693

There are less aggressive ways to do typing like a typealias... but it wouldn't give clear in line coding warnings which is what I think is necessary here with this level of a change. Open to thoughts though.

Reply:
I agree too, also I would say we need to do it more and more, also we're doing it in rust more often.